### PR TITLE
fix some runtime calls reported by JET

### PR DESCRIPTION
**Pull request by KristofferC**
I saw some slowdowns in package loading (not huge but measureable) from `OpenSSL.__init__` getting recompiled due to it getting invalidated. I used JET and tried to fix up some simple stuff. Don't actually know how much effect this will have in practice..

### CODE REVIEW

File: src/OpenSSL.jl [hunk 1]
```diff
@@ -529,15 +529,15 @@ end
 # Locate oss-modules, which contains legacy shared library
 function _ossl_modules_path()
     @static if Sys.iswindows()
-        bin_dir = dirname(OpenSSL_jll.libssl_path)
+        bin_dir = dirname(OpenSSL_jll.libssl_path::String)
```
Comment (KristofferC) [inline src/OpenSSL.jl:532]: ideally, this should  be fixed in JLLWrappers to be inferable. I'll look into that.

Comment (KristofferC) [reply src/OpenSSL.jl:532]: Actually, this is fixed now.

Comment (quinnj) [reply src/OpenSSL.jl:532]: Ok, so we don't need this change anymore? How about the one a few lines below?

Comment (KristofferC) [reply src/OpenSSL.jl:532]: It should be fine to remove both. On 1.9 with JLLWrappers master these should infer to `Union{String, Nothing}`.

Comment (quinnj) [reply src/OpenSSL.jl:532]: kk, I'm going to close the PR then. I believe the other infer problem should already be fixed on main

**Comment by KristofferC** [comment]
For posterity, what I ran was just `using JET, OpenSSL; JET.@report_opt OpenSSL.__init__()`.

**Pull request closed without merging**